Orm/Xtensive.Orm.Tests/Storage/AsyncQueries/ExecuteAsyncExtensionTest.cs

```diff
@@ -0,0 +1,190 @@
+    }
+
+    // [Test]
+    // public async Task DoQueryableAsEnumeableTest()
+    // {
+    //   using (var session = Domain.OpenSession())
+    //   using (var transaction = session.OpenTransaction()) {
+    //     var task = session.Query.All<TestEntity>().Where(e => e.Value < 5 && e.Value > 0).AsEnumerable().AsAsync();
+    //     Assert.That(task.IsCompleted, Is.True);
+    //
+    //     int before = 1;
+    //     foreach (var value in await task) {
+    //       Assert.That(value.Value, Is.EqualTo(before));
+    //       before++;
+    //     }
+    //   }
+    // }
+
+    // [Test]
+    // public async Task PersistentCollectionAsEnumerableTesk()
+    // {
+    //   using (var session = Domain.OpenSession())
+    //   using (var transaction = session.OpenTransaction()) {
+    //     var container = session.Query.All<EntitySetContainer>().First();
+    //     var task = container.EntitySet.Where(e => e.Value < 5 && e.Value > 0).AsEnumerable().AsAsync();
+    //     Assert.That(task.IsCompleted, Is.True);
+    //
+    //     int before = 1;
+    //     foreach (var value in await task) {
+    //       Assert.That(value.Value, Is.EqualTo(before));
+    //       before++;
+    //     }
+    //   }
+    // }
+
+    [Test]
+    public async Task DoQueryableTest()
+    {
+      using (var session = Domain.OpenSession())
+      using (var transaction = session.OpenTransaction()) {
+        var task = session.Query.All<TestEntity>().Where(e => e.Value < 5 && e.Value > 0).ExecuteAsync();
+        Assert.That(task.IsCompleted, Is.False);
+
+        int before = 1;
+        foreach (var value in await task) {
+          Assert.That(value.Value, Is.EqualTo(before));
+          before++;
+        }
+      }
+    }
+
+    [Test]
+    public async Task PersistentCollectionTest()
+    {
+      using (var session = Domain.OpenSession())
+      using (var transaction = session.OpenTransaction()) {
+        var container = session.Query.All<EntitySetContainer>().First();
+        var task = container.EntitySet.Where(e => e.Value < 5 && e.Value > 0).ExecuteAsync();
+        Assert.That(task.IsCompleted, Is.False);
+
+        int before = 1;
+        foreach (var value in await task) {
+          Assert.That(value.Value, Is.EqualTo(before));
+          before++;
+        }
+      }
+    }
+
+    // [Test]
+    // public async Task DelayedQueryTest()
+    // {
+    //   using (var session = Domain.OpenSession())
+    //   using (var transaction = session.OpenTransaction()) {
+    //     var delayed = session.Query.ExecuteDelayed((q) => q.All<TestEntity>().Where(e => e.Value < 5 && e.Value > 0));
+    //     var task = delayed.AsAsync();
+    //     Assert.That(task.IsCompleted, Is.False);
+    //
+    //     int before = 1;
+    //     foreach (var value in await task) {
+    //       Assert.That(value.Value, Is.EqualTo(before));
+    //       before++;
+    //     }
+    //   }
+    // }
+
+    private IEnumerable<T> AsEnumerable<T>(T a, T b, T c, T d)
```

Comment:
Some debug stuff?

Reply:
Fixed.

Reply:
This was a method used by the commented test methods.